Automattic/woocommerce-payments
# Add invalid prodcut id error check

Fixes issue p1699453204196309-slack-C7U3Y3VMY

#### Changes proposed in this Pull Request
Currently, we don't check if the `product_id` is valid before getting the product.  When the `product_id` is not a valid product ID, we'll get a fetal error `Call to a member function get_type() on bool`.

This PR adds an error check when the product_id is invalid and returns the error.

#### Screenshots
<img width="690" alt="Screen Shot 2023-11-08 at 11 34 24 AM" src="https://github.com/Automattic/woocommerce-payments/assets/56378160/786d11f9-a341-4522-a457-5eec471a1a35">

<img width="588" alt="Screen Shot 2023-11-08 at 11 34 35 AM" src="https://github.com/Automattic/woocommerce-payments/assets/56378160/443c1816-22fd-4c41-a9fd-f4308ca1270e">


#### Testing instructions
1. [Comment out this security check](https://github.com/Automattic/woocommerce-payments/blob/d115e3e146853310d6cd23204ca20c76a02a35e8/includes/class-wc-payments-woopay-button-handler.php#L221).
5. Test this POST endpoint `?wc-ajax=wcpay_add_to_cart`
6. Set the content type to `x-www-form-urlencoded`
7. Add `product_id` with a valid product ID, `quantity` with an integer 
8. You should see `"result": "success"`
9. Update the `product_id` to an invalid product id
10. You should not see the fetal error but see `"Invalid product id"` error message

-------------------


**Post merge**
